### PR TITLE
General docs pass

### PR DESCRIPTION
**Pull request by munkyshi**
### Linked issue(s):

None

### What change does this PR introduce and why?

Quick docs pass:

- Update FAQ wording to specify that data files are not uploaded to Kolena
- Clean up cards on index.md pages to surface datasets and not workflows
- Add FAQ entry for working around pandas.DataFrame.to_csv issue
- Add page overview docs for annotation, asset, and io packages

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: docs/faq/index.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ If you don't see your question here, please reach out to us on Slack or at
     We're constantly adding new data types and annotation types — if you don't see what you're looking for, reach out
     to us and we'll happily extend our system to support your use case.
 
-??? faq "Do I have to upload my datasets to Kolena?"
+??? faq "Do I have to upload my data files to Kolena?"
```
Comment (gordonhart) [inline docs/faq/index.md]: I would consider updating to "images, documents, or files"

File: docs/advanced-usage/index.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ hide:
 
 # :kolena-rocket-20: Advanced Usage
 
-This section contains tutorial documentation for advanced features available in Kolena.
+This section contains tutorial documentation for advanced features for Kolena Workflow.
```
Comment (gordonhart) [inline docs/advanced-usage/index.md]: Nit: standardize to "Kolena Workflows" (plural)

**Review by gordonhart — APPROVED** [review]

File: docs/advanced-usage/index.md (hunk 2 above)
Comment (munkyshi) [reply docs/advanced-usage/index.md]: Whoops, this change was supposed to be part of https://github.com/kolenaIO/kolena/pull/431 😅 -- adding to there and renaming to `Kolena Workflows`

File: docs/faq/index.md (hunk 1 above)
Comment (munkyshi) [reply docs/faq/index.md]: Done.

**Pull request merged**